src/components/Dropdown/Dropdown.tsx

```diff
@@ -103,7 +118,11 @@ export function Dropdown(props: PropsWithChildren<DropdownProps>): JSX.Element {
     if (e.key === 'ArrowDown') {
       updateFocusedItem(focusedIndex + 1);
     } else if (e.key === 'ArrowUp') {
-      updateFocusedItem(focusedIndex - 1);
+      if (alwaysSelectOption && value !== focusedValue) {
```

Comment:
if the input box's value is equal to the focused value (e.g. selected something, click out of the text box, and click back in), then arrowing up wouldn't loop to the the last option like you can normally. I'm not sure if that's the behavior you are expecting, but it may be better to have this as `alwaysSelectOption && focusedIndex === 0` so it would still loop regardless